### PR TITLE
Port from nix to rustix

### PR DESCRIPTION
**Pull request by ids1024**
https://github.com/Smithay/wayland-rs/issues/653

Incomplete. Looks like `rustix` needs support for `SO_PEERCRED`.

### CODE REVIEW

**Comment by ids1024** [comment]
See https://github.com/bytecodealliance/rustix/pull/851 for `SO_PEERCRED`.

**Comment by ids1024** [comment]
Now this just needs https://github.com/bytecodealliance/rustix/pull/851 to be merged and released, and I guess I need to test on a BSD to update the kqueue code.

File: wayland-backend/src/rs/socket.rs [hunk 1]
```diff
@@ -141,7 +151,7 @@ impl BufferedSocket {
             let written = self.socket.send_msg(bytes, fds)?;
             for &fd in fds {
                 // once the fds are sent, we can close them
-                let _ = ::nix::unistd::close(fd);
+                let _ = unsafe { libc::close(fd) };
```
Comment (daxpedda) [inline wayland-backend/src/rs/socket.rs:154]: Couldn't [`rustix::io::close()`](https://docs.rs/rustix/0.38.17/rustix/io/fn.close.html) be used here instead?

File: wayland-backend/src/types/server.rs [hunk 2]
```diff
@@ -66,9 +66,9 @@ pub enum DisconnectReason {
...
 pub struct Credentials {
     /// pid of the client
-    pub pid: nix::libc::pid_t,
+    pub pid: libc::pid_t,
     /// uid of the client
-    pub uid: nix::libc::uid_t,
+    pub uid: libc::uid_t,
     /// gid of the client
-    pub gid: nix::libc::gid_t,
+    pub gid: libc::gid_t,
```
Comment (daxpedda) [inline wayland-backend/src/types/server.rs:73]:
```rust
    /// pid of the client
    pub pid: rustix::process::RawPid,
    /// uid of the client
    pub uid: rustix::process::RawUid,
    /// gid of the client
    pub gid: rustix::process::RawGid,
```

**Review by daxpedda — COMMENTED** [review]
I was wondering why `libc` was added here, it seems to me it could be dropped entirely.
(despite the fact it's still in the dependency tree of course)

**Review by elinorbgr — APPROVED** [review]

**Pull request merged**